chromadb/api/models/Collection.py

```diff
@@ -71,17 +72,23 @@ def add(
             ValueError: If you provide an id that already exists
 
         """
-        (
+        embedding_set = self._process_add_request(
             ids,
             embeddings,
             metadatas,
             documents,
+            images,
             uris,
-        ) = self._validate_and_prepare_embedding_set(
-            ids, embeddings, metadatas, documents, images, uris
         )
 
-        self._client._add(ids, self.id, embeddings, metadatas, documents, uris)
+        self._client._add(
+            embedding_set["ids"],
```

Comment:
Maybe we should avoid packing and unpacking stuff into `EmbeddingSet` - that might be a bigger refactor than this change is worth though; let's open an issue.

Reply:
https://github.com/chroma-core/chroma/issues/2705